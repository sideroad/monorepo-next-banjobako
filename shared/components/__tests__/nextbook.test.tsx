import React from 'react';
import NextBook from '../_nextbook';
import renderer from 'react-test-renderer';

test('Button to render document.head', () => {
  const component = renderer.create(
    <>
      {NextBook.books.map(book => (
        <div key={book.name}>
          {book.name}
          {book.component ? book.component : null}
          {book.components
            ? book.components.map(item => (
                <div key={`${book.name}-${item.name}`}>
                  <div>
                    {book.name} - {item.name}
                  </div>
                  <div>{item.component}</div>
                </div>
              ))
            : null}
        </div>
      ))}
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
