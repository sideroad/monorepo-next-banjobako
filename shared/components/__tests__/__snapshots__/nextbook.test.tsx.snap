// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to render document.head 1`] = `
Array [
  <div>
    Button
    <button
      className="jsx-3752195309"
      onClick={[Function]}
    >
      FooBar
    </button>
  </div>,
  <div>
    Link
    <div>
      <div>
        Link
         - 
        Basic usage
      </div>
      <div>
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </div>
    </div>
    <div>
      <div>
        Link
         - 
        With as property
      </div>
      <div>
        <a
          href="/users/sideroad"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          sideroad
        </a>
      </div>
    </div>
  </div>,
]
`;
